### PR TITLE
TYP: tslib.pyi

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -2146,7 +2146,9 @@ def objects_to_datetime64ns(
             require_iso8601=require_iso8601,
             allow_mixed=allow_mixed,
         )
-        result = result.reshape(data.shape, order=order)
+        # error: No overload variant of "reshape" of "_ArrayOrScalarCommon"
+        # matches argument types "Tuple[int, ...]", "str"
+        result = result.reshape(data.shape, order=order)  # type: ignore[call-overload]
```
Comment (simonjayhawkins) [inline pandas/core/arrays/datetimes.py:2151]: added ignore not needed. order is inferred as `str` by mypy. can override the mypy inference with a variable type annotation.

i.e. `order: Literal["F", "C"] = "F" if flags.f_contiguous else "C"`

**Review by simonjayhawkins — APPROVED** [review]
Thanks @jbrockmendel generally lgtm.

File: pandas/core/arrays/datetimes.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/datetimes.py:2151]: wont using Literal here cause problems?  IIUC the `__future__` import makes Literal OK i function annotations, but not variable annotations

File: pandas/_libs/tslib.pyi [hunk 2]
```diff
@@ -0,0 +1,35 @@
+from datetime import tzinfo
+from typing import Optional
+
+import numpy as np
+
+from pandas import Timestamp
+
+def _test_parse_iso8601(ts: str) -> Timestamp: ...
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/_libs/tslib.pyi:8]: does that mean it shouldnt be here?

File: pandas/_libs/tslib.pyi [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def format_array_from_datetime(
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/_libs/tslib.pyi:11]: will add this to my list

File: pandas/_libs/tslib.pyi (hunk 2 above)
[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/_libs/tslib.pyi:8]: I don't think so. I think better in general for mypy to report e.g. `error: Module has no attribute "_test_parse_iso8601"` when accessing the private functions and then adding ignores in tests when testing these private functions.

File: pandas/core/arrays/datetimes.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/core/arrays/datetimes.py:2151]: I _think_ variable annotations (inside functions) aren't parsed by the Python interpreter even without the \_\_future__ import, but will need to confirm this.

Comment (simonjayhawkins) [reply pandas/core/arrays/datetimes.py:2151]: yep. This works.

```
def func():
    a: Unknown
    print("Success!")

func()
```

File: pandas/_libs/tslib.pyi [hunk 4]
```diff
@@ -0,0 +1,35 @@
+from datetime import tzinfo
+from typing import Optional
+
+import numpy as np
+
+from pandas import Timestamp
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/_libs/tslib.pyi]: yep.  this ended up getting remove when i took out _test_parse_iso8601

File: pandas/core/arrays/datetimes.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/datetimes.py]: good call, updated

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**